module/x/gravity/types/ethereum.go

```diff
@@ -66,24 +77,32 @@ func ValidateEthAddress(address string) error {
 	if address == "" {
 		return fmt.Errorf("empty")
 	}
-	if len(address) != ETHContractAddressLen {
-		return fmt.Errorf("address(%s) of the wrong length exp(%d) actual(%d)", address, ETHContractAddressLen, len(address))
+
```

Comment:
The main differences I see here are:
1. address is not required to have a leading "0x", given that the address values stored will always produce a string prefixed by "0x" this should be inconsequential
2. address is iterated twice (once in DecodeString, once in IsHexAddress) checking the characters are within [0-9][a-f][A-F], however DecodeString gves nice errors while IsHexAddress adds on the length check.

Regarding 2. could you change that to just a length check to reduce character iteration?